nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestAuditWithElasticSearch.java

```diff
@@ -244,5 +253,44 @@ public void testGetLogEntriesAfter() throws Exception {
         Assert.assertEquals(id3, entries.get(1).getId());
     }
 
+    @Test
+    public void testSaveAndScroll() throws Exception {
+
+        NXAuditEventsService audit = (NXAuditEventsService) Framework.getRuntime()
+                                                                     .getComponent(NXAuditEventsService.NAME);
+        Assert.assertNotNull(audit);
+
+        ESAuditBackend esBackend = (ESAuditBackend) audit.getBackend();
 
+        List<String> jsonEntries = new ArrayList<>();
+
+        ObjectMapper mapper = new ObjectMapper();
+        for (int i = 0; i < 42; i++) {
+            ObjectNode logEntryJson = mapper.createObjectNode();
+            logEntryJson.put("eventId", "idForAuditStorage");
+            jsonEntries.add(mapper.writeValueAsString(logEntryJson));
+        }
+
+        // Save JSON entries into backend
+        esBackend.append(jsonEntries);
+
+        LogEntryGen.flushAndSync();
+
+        // Query all logs
+        AuditQueryBuilder builder = new AuditQueryBuilder().predicates(Predicates.eq("eventId", "idForAuditStorage"));
+        // builder.predicates()
+        List<LogEntry> logs = esBackend.queryLogs(builder);
+        assertEquals(42, logs.size());
+
+        ScrollResult scrollResult = esBackend.scroll(builder, 5, 10);
+        assertEquals(5, scrollResult.getResultIds().size());
```

Comment:
Could be inside the `while` no?

Reply:
The last scroll results could contain less than 5 entries (in this test for example the last retrieved results will contain 2 entries).

Reply:
Right, so could check count <= 5, or not :smile: